crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -220,6 +292,13 @@ impl ArrayHeap {
             };
             evaluator.opcodes.push(AcirOpcode::Arithmetic(load_on_same_adr));
         }
+
+        let id = array_id.as_u32();
+        evaluator.opcodes.push(AcirOpcode::Block(MemoryBlock {
```

Comment:
Can we use `add_block_opcode` here? Also why do we have an additional memory block here?

Reply:
we should not, nice spot!
I have removed it